scripts/dts/gen_defines.py

```diff
@@ -435,11 +435,23 @@ def irq_name_alias(irq, cell_name):
             alias += "_" + str2ident(cell_name)
         return alias
 
+    def encode_zephyr_multi_level_irq(irq, irq_num):
+        # See doc/reference/kernel/other/interrupts.rst for details
+        # on how this encoding works
+        irq_ctrl = irq.controller
+        while irq_ctrl.interrupts:
+            irq_num = (irq_num + 1) << 8
+            irq_num |= irq_ctrl.interrupts[0].specifier["irq"]
+            irq_ctrl = irq_ctrl.interrupts[0].controller
+        return irq_num
+
     for irq_i, irq in enumerate(dev.interrupts):
         # We ignore the controller for now
         for cell_name, cell_value in irq.specifier.items():
             ident = "IRQ_{}".format(irq_i)
-            if cell_name != "irq":
+            if cell_name == "irq":
+                cell_value = encode_zephyr_multi_level_irq(irq, cell_value)
```

Comment:
Bit icky that this makes the name `cell_value` a lie, but not horrible here...

Reply:
Yeah, didn't like it but felt adding 'if' bellow or some other special handling of "irq" seemed worse.